pkg/engines/tekton/controller.go

```diff
@@ -1,743 +1,138 @@
+		client:       client,
+		logger:       logrus.NewEntry(logrus.StandardLogger()).WithField("controller", controllerName),
+		scheme:       scheme,
+		tektonClient: tektonClient,
+		dashboardURL: dashboardURL,
 	}
 }
 
-// Run actually runs the controller
-func (c *Controller) Run(threadiness int, stopCh <-chan struct{}) error {
-	defer utilruntime.HandleCrash()
-	defer c.queue.ShutDown()
-
-	// Start the informer factories to begin populating the informer caches
-	c.logger.Info("Starting controller")
-
-	defer c.configMapWatcher.Stop()
-
-	// Wait for the caches to be synced before starting workers
-	c.logger.Info("Waiting for informer caches to sync")
-	if ok := cache.WaitForCacheSync(stopCh, c.prSynced, c.lhSynced); !ok {
-		return fmt.Errorf("failed to wait for caches to sync")
-	}
-
-	c.logger.Info("Starting workers")
-	// Launch the appropriate number of workers to process PipelineActivity resources
-	for i := 0; i < threadiness; i++ {
-		go wait.Until(c.runWorker, time.Second, stopCh)
-	}
-
-	c.logger.Info("Started workers")
-	<-stopCh
-	c.logger.Info("Shutting down workers")
-
-	return nil
-}
-
-// runWorker is a long-running function that will continually call the
-// processNextWorkItem function in order to read and process a message on the
-// workqueue.
-func (c *Controller) runWorker() {
-	for c.processNextWorkItem() {
-	}
-}
-
-// processNextWorkItem will read a single work item off the workqueue and
-// attempt to process it, by calling the syncHandler.
-func (c *Controller) processNextWorkItem() bool {
-	obj, shutdown := c.queue.Get()
-
-	if shutdown {
-		return false
-	}
-
-	// We wrap this block in a func so we can defer c.workqueue.Done.
-	err := func(obj interface{}) error {
-		// We call Done here so the workqueue knows we have finished
-		// processing this item. We also must remember to call Forget if we
-		// do not want this work item being re-queued. For example, we do
-		// not call Forget if a transient error occurs, instead the item is
-		// put back on the workqueue and attempted again after a back-off
-		// period.
-		defer c.queue.Done(obj)
-		var key string
-		var ok bool
-		// We expect strings to come off the workqueue. These are of the
-		// form namespace/name. We do this as the delayed nature of the
-		// workqueue means the items in the informer cache may actually be
-		// more up to date that when the item was initially put onto the
-		// workqueue.
-		if key, ok = obj.(string); !ok {
-			// As the item in the workqueue is actually invalid, we call
-			// Forget here else we'd go into a loop of attempting to
-			// process a work item that is invalid.
-			c.queue.Forget(obj)
-			c.logger.Warnf("expected string in workqueue but got %#v", obj)
+// SetupWithManager sets up the reconcilier with it's manager
+func (r *LighthouseJobReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	if err := mgr.GetFieldIndexer().IndexField(&pipelinev1beta1.TaskRun{}, jobOwnerKey, func(rawObj runtime.Object) []string {
```

Comment:
Should this be `PipelineRun` rather than `TaskRun`?

Reply:
Definitely, good catch !

Reply:
That was the main reason behind the issue, thanks :D